pandas/core/arrays/datetimelike.py

```diff
@@ -428,9 +433,26 @@ def astype(self, dtype, copy=True):
         else:
             return np.asarray(self, dtype=dtype)
 
-    def view(self, dtype: Optional[Dtype] = None):
+    def view(self, dtype: Optional[Dtype] = None) -> ArrayLike:
         if dtype is None or dtype is self.dtype:
             return type(self)(self._ndarray, dtype=self.dtype)
+
+        if isinstance(dtype, type):
```

Comment:
i think make this clear that M8[ns] (or really M8[s]) or whatever will hit here, but a tz-aware / pandas dtype will pass thru. yes after i read it i get it, but in 3 months no-one will remember.

Reply:
fleshing out the comment bc its pretty weird.  this line isnt catching np.dtype, its catching _type_, e.g. we get here with `dtype=np.ndarray`

Reply:
updated + green